### PR TITLE
Upgrade to Spring Boot 3.3

### PR DESCRIPTION
**Pull request by paulbakker**
Pull Request type
----

- [ ] Bugfix
- [x ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe):

Changes in this PR
----
Upgrade to Spring Boot 3.3, which comes with graphql-java 22 and federation 5.0.0.

### CODE REVIEW

**Review by iuliiasobolevska — APPROVED** [review]

**Review by Emily — APPROVED** [review]
looks like the main change ends up being the async preparsed document provider? I think this lgtm assuming the test is fixed

**Comment by paulbakker** [comment]
For release notes:
* Mention this bug for Kotlin with `@JsonAnySetter` in Jackson 2.17: https://github.com/FasterXML/jackson-databind/issues/4508

File: graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/ErrorsTest.kt [hunk 1]
```diff
@@ -178,6 +179,7 @@ class ErrorsTest {
     }
 
     @Test
+    @Disabled("Broken by Jackson 2.17 https://github.com/FasterXML/jackson-databind/issues/4508")
```
Comment (kilink) [inline graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/ErrorsTest.kt:182]: There's a workaround in that linked issue that we could apply instead:

https://github.com/FasterXML/jackson-databind/issues/4508#issuecomment-2107622837

Comment (paulbakker) [reply graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/ErrorsTest.kt:182]: @kilink From what I understand from the issue is that the workaround is for the Jackson folks to do, it's not something we can do as a user of Jackson.

**Pull request merged**